src/SimplePie.php

```diff
@@ -3297,7 +3297,7 @@ public function __call(string $method, array $args)
         $trace = debug_backtrace();
         $file = $trace[0]['file'];
         $line = $trace[0]['line'];
-        trigger_error("Call to undefined method $class::$method() in $file on line $line", E_USER_ERROR);
+        throw new Exception("Call to undefined method $class::$method() in $file on line $line");
```

Comment:
> `@throws` probably unnecessary

Oops, forgot to remove the `\` at the start of the name in the `throws` anyway. I'll remove the tag.

> 
> This might need to be `SimplePieException`? Or does the local `Exception` class shadow `\Exception`?

This uses the existing `SimplePie\Exception`. As the classes are in the same namespace, no import `use` statement is needed and the SimplePie `Exception` will be used.

> Or just remove it. The methods have been removed for over twelve years.

Well, yes and no. Those methods have effectively _not_ been removed, but deprecated. Removing them should probably be done in a major version and my intention with this PR is that it will be backported to the 1.8 branch for release ahead of PHP 8.4, so even though `__call()` could be removed in `master`, that wouldn't solve PHP 8.4 compatibility for 1.8.

Reply:
> This uses the existing `SimplePie\Exception`. As the classes are in the same namespace, no import `use` statement is needed and the SimplePie `Exception` will be used.

I would expect that but then why do we have

https://github.com/simplepie/simplepie/blob/9385c54382edec0a768fdbe310c3e580c1b5dc81/src/SimplePie.php#L22

Reply:
> Well, yes and no. Those methods have effectively not been removed, but deprecated.

If they were just deprecated, the `__call` would proxy the call to the original methods. But they were actually both deprecated and removed and the warning was not updated.

Reply:
> > This uses the existing `SimplePie\Exception`. As the classes are in the same namespace, no import `use` statement is needed and the SimplePie `Exception` will be used.
> 
> I would expect that but then why do we have
> 
> https://github.com/simplepie/simplepie/blob/9385c54382edec0a768fdbe310c3e580c1b5dc81/src/SimplePie.php#L22

Very good question, might just be to prevent confusion for contributors ?
It doesn't make a (functional) difference for this PR anyhow, though for consistency, I should maybe update the `throw new` to use the alias.

> If they were just deprecated, the __call would proxy the call to the original methods. But they were actually both deprecated and removed and the warning was not updated.

Fair enough. :+1: Sounds like a good one to clean up in 2.0 in that case (after this PR has been merged and backported to 1.8).

Reply:
Done.